h2/connection.py

```diff
@@ -586,8 +593,11 @@ def initiate_upgrade_connection(self, settings_header=None):
             For servers, returns nothing.
         :rtype: ``bytes`` or ``None``
         """
-        frame_data = None
+        self.config.logger.debug(
+            "Upgrade connection.\nSettings frame: %s", self.local_settings
```

Comment:
👎 for multi-line log messages.

One line per log makes things way easier to parse the logs with automated tools. Extra lines just cause confusion and/or get lost.